### PR TITLE
fix(api): register service check if already exists based on consumer

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12170.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✘

File: engine/api/services.go [hunk 1]
```diff
@@ -97,6 +86,8 @@ func (api *API) postServiceRegisterHandler() service.Handler {
 			return sdk.WithStack(err)
 		}
 
+		data.Uptodate = data.Version == sdk.VERSION
```
Comment (fsamin) [inline engine/api/services.go]:
```suggestion
		srv.Uptodate = data.Version == sdk.VERSION
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12173.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12173.1 ■   
* Build
  * Build and Package All ■   
  * Build Debpacker ■   
  * Build OS-Ansible-Inventory ■   
  * Build SMTPMock ■   
  * Lint UI Translate ■   
  * Test CLI ■   
  * Test Engine ■   
  * Test SDK ■   
  * Test UI ■

**Pull request merged**